### PR TITLE
Upgrade rn-builder-bob to enable tree-shaking in rn-web

### PR DESCRIPTION
**Pull request by merrywhether**
# Summary

The `.web.js` path in this library currently [breaks tree-shaking in web projects](https://github.com/callstack/react-native-builder-bob/issues/154), causing them to import the entire ~250kb of react-native-web regardless of their other configuration: 
Screenshot 2023-02-24 at 6 06 26 PM

This is because the module build artifacts like `lib/module/ReactNativeSVG.web.js` are in pure ES5 and not ES5+ESM (es5 but preserving import/export statements).

Screenshot 2023-02-24 at 5 55 26 PM

Updating to the latest version of `rn-builder-bob` (which had a prerequisite fix in <https://github.com/callstack/react-native-builder-bob/pull/363> for this library) fixes this, generating the proper ES5+ESM output for the module build (while leaving the commonjs build in pure ES5 with `require`):

Screenshot 2023-02-24 at 5 53 22 PM
Screenshot 2023-02-24 at 6 23 36 PM

## Test Plan

This change should only affect the build output and not anything in the native layers.

Verified against simulators:
image

### What's required for testing (prerequisites)?

n/a

### What are the steps to reproduce (after prerequisites)?

Generally navigate around and see that everything still works.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] (n/a) I added documentation in `README.md`
- [x] (n/a) I updated the typed files (typescript)
- [x] (n/a) I added a test for the API in the `__tests__` folder

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,10 +70,7 @@
     "css-tree": "^1.1.3"
   },
   "devDependencies": {
-    "@babel/core": "^7.18.2",
-    "@babel/plugin-syntax-flow": "^7.17.12",
-    "@babel/plugin-transform-react-jsx": "^7.17.12",
```
Comment (merrywhether) [inline package.json:75]: These are provided by RN/metro already.

**Comment by merrywhether** [comment]
Looked into the failing builds and realized I'd pulled from our private registry at first. Regenerated `yarn.lock` changes from public registry.

**Review by WoLewicki — APPROVED** [review]
LGTM! Thanks for your contribution!

**Comment by merrywhether** [comment]
> @merrywhether Could you also bump the version of node used in the JS build workflow? Based on the failed build it should be 16 here:
> 
> https://github.com/software-mansion/react-native-svg/blob/d6975ea9f20f2fefb603f2612fdb3bbc7bf071ca/.github/workflows/js-build-test.yml#L21
> 
> and it the step name.
> Ok I edited the PR and all checks pass now 🚀

Thanks for bumping that. Sorry I was a little slow on the follow-up responsiveness here but happy that we got it merged.

**Comment by WoLewicki** [comment]
Yeah no problem 😅

**Pull request merged**